pkg/templates/externalccm/digitalocean.go

```diff
@@ -0,0 +1,262 @@
+			},
+			{
+				APIGroups: []string{""},
+				Resources: []string{"endpoints"},
+				Verbs:     []string{"create", "get", "list", "watch", "update"},
+			},
+		},
+	}
+}
+
+func doClusterRoleBinding() *rbacv1.ClusterRoleBinding {
+	return &rbacv1.ClusterRoleBinding{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "rbac.authorization.k8s.io/v1",
+			Kind:       "ClusterRoleBinding",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "system:cloud-controller-manager",
+		},
+		RoleRef: rbacv1.RoleRef{
+			APIGroup: rbacv1.GroupName,
+			Name:     "system:cloud-controller-manager",
+			Kind:     "ClusterRole",
+		},
+		Subjects: []rbacv1.Subject{
+			{
+				Kind:      "ServiceAccount",
+				Name:      digitaloceanSAName,
+				Namespace: metav1.NamespaceSystem,
+			},
+		},
+	}
+}
+
+func doDeployment() *appsv1.Deployment {
+	var (
+		replicas  int32 = 1
+		revisions int32 = 2
+	)
+
+	return &appsv1.Deployment{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "apps/v1",
+			Kind:       "Deployment",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      digitaloceanDeploymentName,
+			Namespace: metav1.NamespaceSystem,
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas:             &replicas,
+			RevisionHistoryLimit: &revisions,
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"app": "digitalocean-cloud-controller-manager",
+				},
+			},
+			Strategy: appsv1.DeploymentStrategy{
+				Type: appsv1.RecreateDeploymentStrategyType,
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Annotations: map[string]string{
+						"scheduler.alpha.kubernetes.io/critical-pod": "",
+					},
+					Labels: map[string]string{
+						"app": "digitalocean-cloud-controller-manager",
+					},
+				},
+				Spec: corev1.PodSpec{
+					ServiceAccountName: digitaloceanSAName,
+					Tolerations: []corev1.Toleration{
+						{
+							Key:      "node-role.kubernetes.io/master",
+							Operator: corev1.TolerationOpExists,
+							Effect:   corev1.TaintEffectNoSchedule,
+						},
+						{
+							Key:    "node.cloudprovider.kubernetes.io/uninitialized",
+							Value:  "true",
+							Effect: corev1.TaintEffectNoSchedule,
+						},
+						{
+							Key:      "CriticalAddonsOnly",
+							Operator: corev1.TolerationOpExists,
+						},
+					},
+					Containers: []corev1.Container{
+						{
+							Name:  "digitalocean-cloud-controller-manager",
+							Image: "digitalocean/digitalocean-cloud-controller-manager:" + digitaloceanCCMVersion,
+							Command: []string{
+								"/bin/digitalocean-cloud-controller-manager",
+								"--cloud-provider=digitalocean",
+								"--leader-elect=false",
+							},
+							Env: []corev1.EnvVar{
+								{
+									Name: "DO_ACCESS_TOKEN",
+									ValueFrom: &corev1.EnvVarSource{
+										SecretKeyRef: &corev1.SecretKeySelector{
+											LocalObjectReference: corev1.LocalObjectReference{
+												Name: machinecontroller.MachineControllerCredentialsSecretName,
```

Comment:
This is not going to work if the `machine-controller` isn't deployed. We should export that logic from the `machine-controller` and deploy it regardless.

Reply:
We will handle this in separate PR: https://github.com/kubermatic/kubeone/issues/366

Reply:
I'll take care of it with the API changes.